src/codec.rs

```diff
@@ -1082,7 +1084,7 @@ where
 
 		decode_chunk(&mut decoded_vec, chunk_len)?;
 
-		num_undecoded_items = num_undecoded_items.saturating_sub(chunk_len);
+		num_undecoded_items -= chunk_len;
```

Comment:
I think for such critical code its fine to stay a bit defensive with `saturating` or `checked` even if it *should* not happen.

Reply:
Personally I agree. I would stay on the safe side. @koute  WDYT ?

Reply:
We already do the `saturated_sub` by doing the `chunk_len.min(num_undecoded_items)` above.

Reply:
I see it more like a safety measure., In case someone changes the logic sometime in the future, and the min disappears somehow, but forgets to then also use `saturated_sub`. But I agree that it's a very unlikely scenario. I'm ok with keeping it as it is.

Reply:
In general I don't necessarily like including useless code for extra "safety" as it muddles the intent of the code, and then whoever is reading is left scratching their head why that code is there.

If we need to ensure that something is true then I prefer, in no particular order: a) adding extra asserts, b) adding tests, c) producing and enforcing a proof that the condition is impossible (quite easy to do in a lot of cases with something like [Kani](https://github.com/model-checking/kani), which is essentially black magic).